pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -242,12 +231,13 @@ func handleStatusGen(
 // Otherwise it will add our finalizer if it is absent.
 func handleFinalizers(
```

Comment:
Some changes to this function proposed in my long comment.

Reply:
Done.